src/app/components/playback/playback_control.rs

```diff
@@ -51,6 +51,16 @@ impl PlaybackControlModel {
         self.dispatcher.dispatch(PlaybackAction::TogglePlay.into());
     }
 
+    pub fn toggle_shuffle(&self) {
+        if self.is_playing() {
```

Comment:
is this actually needed? if yes, should toggle_repeat have a similar guard?

Reply:
I decided to add this for shuffle because when attempting to create a playlist with in shuffle, the shuffle function in playback_state.rs has no current song. This caused the song index to be random.

Reply:
hmm I think I see what you mean, but this should be something that playback_state has to handle better then :) (not saying you have to fix it!) the guard can be remove for now

Also is_playing is not what you think it is, it returns false when playback is paused :/

Reply:
I have decided to remove the guard and change play to handle the special case that caused the issue. Now it will swap when needed and keep the shuffle consistency